addons/better-img-uploads/userscript.js

```diff
@@ -0,0 +1,101 @@
+export default async function ({ addon, console }) {
+  let enabled = true; //Addon is enabled
+
+  addon.self.addEventListener("disabled", () => (enabled = false));
+  addon.self.addEventListener("reenabled", () => console.log((enabled = true)));
+
+  while (true) {
+    let ignore = false; //Have files already been changed?
+
+    let input = await addon.tab.waitForElement('input[type="file"][accept*=".svg"]', {
+      markAsSeen: true,
+    }); //Wait for the next image file input
+
+    input.addEventListener(
+      "change",
+      (e) => {
+        if (!enabled) return;
+        if (!ignore) (ignore = true) && onchange(e);
+        //Files have not been changed yet...
+        else ignore = false;
+      },
+      true
+    );
+  }
+
+  async function onchange(e) {
+    e.stopPropagation(); //Prevent Scratch from seeing the event so that we can process the images first
+
+    let el = e.target;
+    let files = Array.from(el.files);
+    let processed = new Array();
+
+    for (let file of files) {
+      if (file.type.includes("svg")) {
+        //The file is already a svg, we should not change it...
+        processed.push(file);
+        continue;
+      }
+
+      let blob = await new Promise((resolve) => {
```

Comment:
```js
let blob = file;
```

Reply:
That doesn't work, just throws an error...

Reply:
In order to correctly add the image to an svg for scratch it has to be a dataURL

Reply:
ok